host/windows/syscall.c

```diff
@@ -480,142 +948,469 @@ ssize_t oe_syscall_writev_ocall(
+        goto done;
+    }
+
+done:
+    if (wpathname)
+    {
+        free(wpathname);
+    }
+    return ret;
 }
 
 int oe_syscall_link_ocall(const char* oldpath, const char* newpath)
 {
-    OE_UNUSED(oldpath);
-    OE_UNUSED(newpath);
+    int ret = -1;
+    WCHAR* woldpath = oe_syscall_path_to_win(oldpath, NULL);
+    WCHAR* wnewpath = oe_syscall_path_to_win(newpath, NULL);
 
-    PANIC;
+    if (!CreateHardLinkW(wnewpath, woldpath, NULL))
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
+    ret = 0;
+
+done:
+    if (woldpath)
+    {
+        free(woldpath);
+    }
+
+    if (wnewpath)
+    {
+        free(wnewpath);
+    }
+    return ret;
 }
 
 int oe_syscall_unlink_ocall(const char* pathname)
 {
-    OE_UNUSED(pathname);
+    int ret = -1;
+    WCHAR* wpathname = oe_syscall_path_to_win(pathname, NULL);
 
-    PANIC;
+    ret = _wunlink(wpathname);
+    if (ret < 0)
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
+
+done:
+    if (wpathname)
+    {
+        free(wpathname);
+    }
+    return ret;
 }
 
 int oe_syscall_rename_ocall(const char* oldpath, const char* newpath)
 {
-    OE_UNUSED(oldpath);
-    OE_UNUSED(newpath);
+    int ret = -1;
+    WCHAR* woldpath = oe_syscall_path_to_win(oldpath, NULL);
+    WCHAR* wnewpath = oe_syscall_path_to_win(newpath, NULL);
 
-    PANIC;
+    ret = _wrename(woldpath, wnewpath);
+    if (ret < 0)
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
+
+done:
+    if (woldpath)
+    {
+        free(woldpath);
+    }
+
+    if (wnewpath)
+    {
+        free(wnewpath);
+    }
+    return ret;
 }
 
 int oe_syscall_truncate_ocall(const char* pathname, oe_off_t length)
 {
-    OE_UNUSED(pathname);
-    OE_UNUSED(length);
+    int ret = -1;
+    DWORD sfp_rtn = 0;
+    LARGE_INTEGER new_offset = {0};
+    WCHAR* wpathname = oe_syscall_path_to_win(pathname, NULL);
+
+    HANDLE h = CreateFileW(
+        wpathname,
+        GENERIC_WRITE,
+        FILE_SHARE_WRITE,
+        NULL,
+        OPEN_EXISTING,
+        FILE_ATTRIBUTE_NORMAL,
+        NULL);
+    if (h == INVALID_HANDLE_VALUE)
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
 
-    PANIC;
+    new_offset.QuadPart = length;
+    if (!SetFilePointerEx(
+            h, new_offset, (PLARGE_INTEGER)&new_offset, FILE_BEGIN))
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
+
+    if (!SetEndOfFile(h))
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
+
+    CloseHandle(h);
+
+    ret = 0;
+
+done:
+    if (wpathname)
+    {
+        free(wpathname);
+    }
+    return ret;
 }
 
 int oe_syscall_mkdir_ocall(const char* pathname, oe_mode_t mode)
 {
-    OE_UNUSED(pathname);
-    OE_UNUSED(mode);
+    int ret = -1;
+    WCHAR* wpathname = oe_syscall_path_to_win(pathname, NULL);
 
-    PANIC;
+    ret = _wmkdir(wpathname);
```

Comment:
The mode is unused. Is this the intent?

Reply:
Windows doesn't do mode in the same way as linux and can only take the first part which is the attribute for owner. But readable, writable and executable attributes are obvious necessary. Thus mode is useless as an input.